instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/decorator/BaseDecorator.java

```diff
@@ -29,19 +32,26 @@
 import java.lang.reflect.Method;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
-import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ExecutionException;
 
 @Deprecated
 public abstract class BaseDecorator {
 
-  private static final ClassValue<SpanNames> SPAN_NAMES =
-      new ClassValue<SpanNames>() {
-        @Override
-        protected SpanNames computeValue(Class<?> type) {
-          return new SpanNames(getClassName(type));
-        }
-      };
+  private static final QualifiedClassNameCache CLASS_NAMES =
+      new QualifiedClassNameCache(
+          new Function<Class<?>, String>() {
+            @Override
+            public String apply(Class<?> clazz) {
+              String simpleName = clazz.getSimpleName();
+              if (simpleName.isEmpty()) {
+                String name = clazz.getName();
+                int start = name.lastIndexOf('.');
```

Comment:
Is this different from `getEnclosingClass().getSimpleName()` (looping up until hit)? If so, I think this has a good chance of beating the cache in cases except for extremely deeply nested anonymous classes.

Reply:
Nope.

Reply:
This produces names like `EnclosingClass$1` and does this to maintain behaviour. This code is not new and was moved here in a refactoring.